physical/dynamodb/dynamodb.go

```diff
@@ -163,13 +163,16 @@ func NewDynamoDBBackend(conf map[string]string, logger log.Logger) (physical.Bac
 	if endpoint == "" {
 		endpoint = conf["endpoint"]
 	}
-	region := os.Getenv("AWS_REGION")
+	region := os.Getenv("AWS_DYNAMODB_REGION")
```

Comment:
@swenson I wonder if using a `switch` is cleaner instead of deep nesting. I know that we have this pattern for the other config params, but since we're evaluating several env vars for this one it might not be a bad idea?
```go
	var region string
	switch {
	case os.Getenv("AWS_DYNAMODB_REGION") != "":
		region = os.Getenv("AWS_DYNAMODB_REGION")
	case os.Getenv("AWS_REGION") != "":
		region = os.Getenv("AWS_REGION")
	case os.Getenv("AWS_DEFAULT_REGION") != "":
		region = os.Getenv("AWS_DEFAULT_REGION")
	case conf["region"] != "":
		region = conf["region"]
	default:
		region = DefaultDynamoDBRegion
	}
```

Reply:
Oh, I was thinking that the deep nesting was a little bit much. The switch is much cleaner.